modules/gapi/include/opencv2/gapi/infer.hpp

```diff
@@ -121,6 +121,47 @@ struct GInferBase {
     }
 };
 
+struct InOutInfo
+{
+    std::vector<std::string> in_names;
+    std::vector<std::string> out_names;
+};
+
+class GInferInputs
+{
+public:
+    cv::GMat& operator[](const std::string& name) { return in_blobs[name]; }
+    const std::unordered_map<std::string, cv::GMat>& getBlobs() const { return in_blobs; }
+
+private:
+    std::unordered_map<std::string, cv::GMat> in_blobs;
+};
+
+struct GInferOutputs
+{
+public:
+    GInferOutputs(std::shared_ptr<cv::GCall> call)
+        : m_call(std::move(call)), m_info(cv::util::any_cast<InOutInfo>(&m_call->params()))
```

Comment:
what's the reason to move a shared_ptr? It is shared anyway isn't it?

Reply:
1) For logical reason: I don't want to share object between `GInferOutputs` and `infer`
2) It works faster

Reply:
You will share it anyway, don' you?

Reply:
No, I don't do that, I move the object, only one instance exists at a time.

Reply:
You move, but your input is value. This `GCall` remains somewhere in the context of caller.